### ISSUE TITLE
Support Cloudflare Tunnels (Remote) Configuration API

### ISSUE THREAD

**Issue by jtcressy** [cloudflare/cloudflare-go#879]
### Current cloudflare-go version

v0.38.0

### Description

https://api.cloudflare.com/#cloudflare-tunnel-configuration-properties

There do not seem to be any issues discussing support for the `configurations` subresource for cfd tunnels, and I feel like this is a very important feature to support in this library.

### Use cases

End goal would be to use cloudflare-go to provide a programmatic method for administrating _remote tunnel configurations_

I was very excited to see this feature pop up as part of the zero trust product because that means we can configure our cloudflared tunnel connectors with a single secret, and not a stateful config file. It just vastly simplifies usage of cloudflared and how it gets deployed.

Once cloudflare-go supports this API, I hope to see it pop up in the terraform provider and other golang-based projects.

### Potential cloudflare-go usage

```go
https://api.cloudflare.com/#cloudflare-tunnel-configuration-properties
```

### References

https://api.cloudflare.com/#cloudflare-tunnel-configuration-properties

**Comment by Cyb3r-Jak3** [comment]
I can work on this

**Comment by jacobbednarz** [comment]
closed by #948

**Comment by luckman212** [comment]
@jacobbednarz Is there an equivalent flag for [`flarectl`](https://github.com/cloudflare/cloudflare-go/tree/master/cmd/flarectl) that can be used to check zero trust tunnel status via CLI?
